File: Middleware/api/api/Models/BindingModels/CustomerRegistrationBindingModel.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace api.Models.BindingModels
{
    public class CustomerRegistrationBindingModel
    {
        [JsonProperty("emailAddress")]
        [JsonRequired]
        public string emailAddress { get; set; }

        [JsonProperty("customerPassword")]
        [JsonRequired]
        public string customerPassword { get; set; }

        [JsonProperty("firstName")]
        [JsonRequired]
        public string firstName { get; set; }

        [JsonProperty("lastName")]
        [JsonRequired]
        public string lastName { get; set; }

        [JsonProperty("dateOfBirth")]
        [JsonRequired]
        public DateTime dateOfBirth { get; set; }

        [JsonProperty("addressLineOne")]
        [JsonRequired]
        public string addressLineOne { get; set; }

        [JsonProperty("addressLineTwo")]
        public string addressLineTwo { get; set; }

        [JsonProperty("postCode")]
        [JsonRequired]
        public string postCode { get; set; }

        [JsonProperty("mobileNumber")]
        [JsonRequired]
        public string mobileNumber { get; set; }
    }
}

File: Mobile Application/app/src/main/java/com/example/customermobileapplication/BindingModels/CustomerRegistrationBindingModel.java
package com.example.customermobileapplication.BindingModels;

import com.google.gson.annotations.SerializedName;

import java.util.Date;

public class CustomerRegistrationBindingModel {

    @SerializedName("emailAddress")
    public String emailAddress;

    @SerializedName("customerPassword")
    public String customerPassword;

    @SerializedName("firstName")
    public String firstName;

    @SerializedName("lastName")
    public String lastName;

    @SerializedName("dateOfBirth")
    public String dateOfBirth;

    @SerializedName("addressLineOne")
    public String addressLineOne;

    @SerializedName("addressLIneTwo")
    public String addressLineTwo;

    @SerializedName("postCode")
    public String postCode;

    @SerializedName("mobileNumber")
    public String mobileNumber;

    public String getEmailAddress() {
        return emailAddress;
    }

    public void setEmailAddress(String emailAddress) {
        this.emailAddress = emailAddress;
    }

    public String getCustomerPassword() {
        return customerPassword;
    }

    public void setCustomerPassword(String customerPassword) {
        this.customerPassword = customerPassword;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getDateOfBirth() {
        return dateOfBirth;
    }

    public void setDateOfBirth(String dateOfBirth) {
        this.dateOfBirth = dateOfBirth;
    }

    public String getAddressLineOne() {
        return addressLineOne;
    }

    public void setAddressLineOne(String addressLineOne) {
        this.addressLineOne = addressLineOne;
    }

    public String getAddressLineTwo() {
        return addressLineTwo;
    }

    public void setAddressLineTwo(String addressLineTwo) {
        this.addressLineTwo = addressLineTwo;
    }

    public String getPostCode() {
        return postCode;
    }

    public void setPostCode(String postCode) {
        this.postCode = postCode;
    }

    public String getMobileNumber() {
        return mobileNumber;
    }

    public void setMobileNumber(String mobileNumber) {
        this.mobileNumber = mobileNumber;
    }
}
